wealth-management-client/src/app/modal/stock.ts

```diff
@@ -1,25 +1,35 @@
 export class Stock {
   script: string;
   name: string;
-  buyPrice: number;
+  investPrice: number;
   targetPrice: number;
   rcomPrice: number;
   rcomDate: Date;
   rcomTimeScale: string;
   rcomBy: string;
+  rtargetPercentage: number;
+  days: number;
 
   constructor() {
 
   }
 
-  setData(data: any) : void {
+  setData(data: any): void {
+    var firstDate = new Date(data.rcomDate);
+    var secondDate = new Date();
     this.script = data.script;
     this.name = data.name;
-    this.buyPrice = data.buyPrice;
-    this.targetPrice = data.targetPrice;
+    this.investPrice = data.buyPrice;
+
     this.rcomPrice = data.rcomPrice;
-    this.rcomDate = data.rcomDate;
+    this.targetPrice = data.targetPrice;
+    this.rtargetPercentage = (((data.targetPrice - data.rcomPrice) / data.rcomPrice) * 100).toFixed(2);
+
+    this.rcomDate = new Date(data.rcomDate);
+    this.days = Math.abs((firstDate.valueOf() - secondDate.valueOf()) / (24 * 60 * 60 * 1000)).toFixed(0);
```

Comment:
@coderhindu  is the right design to use pipes?

Reply:
Yes, view changes can be handled in pipes. No offence doing here.